samples/testing/states/test_states.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+import qsharp
+
+@pytest.fixture(autouse=True)
+def setup():
+    """Fixture to execute before a test is run"""
+    # Setting the project root to current folder.
+    qsharp.init(project_root=".")
+    yield # this is where the testing happens
+
+# 1/2 (|00⟩ + i|01⟩ - |10⟩ - i|11⟩)
+# The basis states are converted to indices of amplitudes in the array using big endian notation.
+expected_state = [0.5, 0.5j, -0.5, -0.5j]
+
+def test_state_exact() -> None:
+    """Test that Q# code prepares the expected state exactly using Python test code."""
+    # Run Q# code that allocates the qubits and prepares the state but doesn't deallocate the qubits.
+    qsharp.eval(f"use qs = Qubit[2]; StatePrep.PrepareStateWithComplexPhases(qs);")
+    # Get the state of the allocated qubits and convert it to a dense vector.
+    state = qsharp.dump_machine().as_dense_state()
+    # Compare two vectors.
+    assert state == pytest.approx(expected_state)
+    # If you use PrepareStateWithGlobalPhase operation in this test instead, the test will fail,
+    # since the prepared state differs from the expected one by a global phase.
```

Comment:
Rather than saying the test will fail with the other method, you could have a second test case that asserts they are not equal. Something like:
```python
def test_state_exact_rejects_global_phase() -> None:
    """Test that Q# code prepares the expected state exactly using Python test code."""
    # Run Q# code that allocates the qubits and prepares the state but doesn't deallocate the qubits.
    qsharp.eval(f"use qs = Qubit[2]; StatePrep.PrepareStateWithGlobalPhase(qs);")
    # Get the state of the allocated qubits and convert it to a dense vector.
    state = qsharp.dump_machine().as_dense_state()
    # Compare two vectors. Here we expect them to fail equality due to global phase.
    assert state != pytest.approx(expected_state)
```

Reply:
Good point! I'm so used to checking that the state _is_ as expected, didn't think about a test to show the other way :-)